src/lib/y2storage/proposal_settings.rb

```diff
@@ -142,204 +237,185 @@ def delete_forced(type)
-      set_from_boolean_feature(:use_snapshots, :proposal_snapshots)
-      set_from_boolean_feature(:enlarge_swap_for_suspend, :swap_for_suspend)
-      set_from_boolean_feature(:root_subvolume_read_only, :root_subvolume_read_only)
-
-      set_from_size_feature(:root_base_size, :root_base_size)
-      set_from_size_feature(:root_max_size, :root_max_size)
-      set_from_size_feature(:home_max_size, :vm_home_max_size)
-      set_from_size_feature(:min_size_to_use_separate_home, :limit_try_home)
-
-      set_from_integer_feature(:root_space_percent, :root_space_percent)
-      set_from_integer_feature(:btrfs_increase_percentage, :btrfs_increase_percentage)
-
-      set_from_string_feature(:btrfs_default_subvolume, :btrfs_default_subvolume)
-      read_subvolumes_section!
+    # Sets default values for the settings.
+    # These will be the final values when the setting is not specified in the control file
+    def apply_defaults
+      ng_format? ? apply_ng_defaults : apply_legacy_defaults
     end
 
-    # New object initialized according to the YaST product features
-    # (i.e. /control.xml)
+    # Overrides the settings with values read from the YaST product features
+    # (i.e. values in /control.xml).
     #
-    # @return [ProposalSettings]
-    def self.new_for_current_product
-      settings = new
-      settings.read_product_features!
-      settings
+    # Settings omitted in the product features are not modified.
+    def load_features
+      ng_format? ? load_ng_features : load_legacy_features
     end
 
-    def to_s
-      "Storage ProposalSettings\n" \
-        "  use_lvm: #{use_lvm}\n" \
-        "  root_filesystem_type: #{root_filesystem_type}\n" \
-        "  use_snapshots: #{use_snapshots}\n" \
-        "  use_separate_home: #{use_separate_home}\n" \
-        "  home_filesystem_type: #{home_filesystem_type}\n" \
-        "  enlarge_swap_for_suspend: #{enlarge_swap_for_suspend}\n" \
-        "  root_device: #{root_device}\n" \
-        "  candidate_devices: #{candidate_devices}\n" \
-        "  root_base_size: #{root_base_size}\n" \
-        "  root_max_size: #{root_max_size}\n" \
-        "  root_space_percent: #{root_space_percent}\n" \
-        "  btrfs_increase_percentage: #{btrfs_increase_percentage}\n" \
-        "  min_size_to_use_separate_home: #{min_size_to_use_separate_home}\n" \
-        "  btrfs_default_subvolume: #{btrfs_default_subvolume}\n" \
-        "  root_subvolume_read_only: #{root_subvolume_read_only}\n" \
-        "  home_min_size: #{home_min_size}\n" \
-        "  home_max_size: #{home_max_size}\n" \
-        "  subvolumes: \n#{subvolumes}\n"
+    def apply_ng_defaults
+      self.lvm                           ||= false
+      self.resize_windows                ||= true
+      self.windows_delete_mode           ||= :ondemand
+      self.linux_delete_mode             ||= :ondemand
+      self.other_delete_mode             ||= :ondemand
+      self.lvm_vg_strategy               ||= :use_needed
```

Comment:
So far this is the only implemented strategy. But in the future I believe it should not be the default, because I'm pretty sure it's going to be by far the slowest one and because is not what we currently do in SLE or openSUSE. `:use_available` would be a more sensible default, I would say.

Reply:
Maybe a #NOTE or #FIXME comment?